router-tests/mcp_test.go

```diff
@@ -553,4 +555,126 @@ func TestMCP(t *testing.T) {
 			})
 		})
 	})
+
+	t.Run("Header Forwarding", func(t *testing.T) {
+		t.Run("MCP server forwards all headers including custom headers like foo:bar", func(t *testing.T) {
```

Comment:
The title sounds weird. Why is foo:bar different from other headers when we use wildcard?